src/briefcase/bootstraps/base.py

```diff
@@ -0,0 +1,142 @@
+from __future__ import annotations
+
+from abc import ABC
+from typing import Literal, TypedDict
+
+
+class ContextDict(TypedDict):
+    formal_name: str
+    app_name: str
+    class_name: str
+    module_name: str
+    project_name: str
+    description: str
+    author: str
+    author_email: str
+    bundle: str
+    url: str
+    license: str
+
+
+class BaseGuiPlugin(ABC):
+    """Definition for a plugin that defines a new Briefcase app."""
+
+    name: str
+    fields: list[str] = [
+        "app_source",
+        "start_app_source",
+        "requires",
+        "macos_requires",
+        "macos_universal_build",
+        "linux_requires",
+        "linux_system_debian_system_requires",
+        "linux_system_debian_system_runtime_requires",
+        "linux_system_rhel_system_requires",
+        "linux_system_rhel_system_runtime_requires",
+        "linux_system_suse_system_requires",
+        "linux_system_suse_system_runtime_requires",
+        "linux_system_arch_system_requires",
+        "linux_system_arch_system_runtime_requires",
+        "linux_appimage_manylinux",
+        "linux_appimage_system_requires",
+        "linux_appimage_linuxdeploy_plugins",
+        "linux_flatpak_runtime",
+        "linux_flatpak_runtime_version",
+        "linux_flatpak_sdk",
+        "windows_requires",
+        "ios_requires",
+        "ios_supported",
+        "android_requires",
+        "android_supported",
+        "web_requires",
+        "web_supported",
+        "web_style_framework",
```

Comment:
This one should hopefully disappear - #1166 includes code that makes the style framework used by web deployments a feature of the GUI toolkit.

However that raises another question - details in the overall comment.